cat-launcher/src-tauri/src/infra/github/api.rs

```diff
@@ -0,0 +1,34 @@
+use crate::infra::github::types::GitHubCommit;
+use reqwest::Client;
+
+#[derive(thiserror::Error, Debug)]
+pub enum GetLatestCommitError {
+  #[error("failed to make request to github api: {0}")]
+  GitHubApiRequest(#[from] reqwest::Error),
+
+  #[error("failed to parse github url `{0}`")]
+  ParseGithubUrl(String),
+}
+
+pub async fn get_latest_commit(
+  github_url: &str,
```

Comment:
Don't accept generic arguments for API functions. Accept exact arguments, such as repo.